packages/job-manager/src/batch/batch.spec.ts

```diff
@@ -65,191 +83,97 @@ describe(Batch, () => {
-                    state: JobTaskState.completed,
-                    result: JobTaskExecutionResult.success,
-                    correlationId: 'messageId-1',
-                },
-                {
-                    id: 'job-2',
-                    state: JobTaskState.completed,
-                    result: JobTaskExecutionResult.success,
-                    correlationId: 'messageId-2',
-                },
-            ];
-            let i = 0;
-            const messages = [
-                {
-                    messageId: 'messageId-1',
-                    messageText: '{}',
-                },
-                {
-                    messageId: 'messageId-2',
-                    messageText: '{}',
-                },
-            ];
-
-            taskMock
-                .setup(async o => o.list(jobId1))
-                .returns(async () => {
-                    if (i++ === 0) {
-                        return Promise.resolve(cloudTaskListResultFirst as any);
-                    } else {
-                        return Promise.resolve(<BatchServiceModels.CloudTaskListResult>{});
-                    }
-                })
-                .verifiable();
-            taskMock
-                .setup(async o => o.listNext('nextPageLink'))
-                .returns(async () => Promise.resolve(cloudTaskListResultNext as BatchServiceModels.TaskListResponse))
-                .verifiable();
-            taskMock
-                .setup(async o => o.addCollection(jobId1, It.isAny()))
-                .returns(async () => Promise.resolve({ value: [] } as BatchServiceModels.TaskAddCollectionResponse));
-
-            const jobTask = await batch.createTasks(jobId1, messages);
-            cloudTaskListResultFirst[0].id = jobTask[0].id;
-            cloudTaskListResultNext[0].id = jobTask[1].id;
-            jobTasksExpected[0].id = jobTask[0].id;
-            jobTasksExpected[1].id = jobTask[1].id;
-
-            const tasksActual = await batch.getCreatedTasksState(jobId1);
+    describe('createTasks()', () => {
+        it('create no new tasks when no messages provided', async () => {
+            const messages: Message[] = [];
+            const tasksActual = await batch.createTasks(jobId1, messages);
 
-            expect(tasksActual).toEqual(jobTasksExpected);
-            taskMock.verifyAll();
+            expect(tasksActual.length).toBe(0);
         });
 
-        it('verifies taskAddParameter data', async () => {
-            const cloudTaskListResultFirst = <BatchServiceModels.CloudTaskListResult>[];
-            cloudTaskListResultFirst.push({
-                id: jobId1,
-                state: JobTaskState.completed,
-                executionInfo: <BatchServiceModels.TaskExecutionInformation>{
-                    result: JobTaskExecutionResult.success,
-                },
-            });
+        it('should add no more than 100 tasks in a single Batch API call', async () => {
+            const messagesCount = 150;
```

Comment:
> messagesCount = 150; [](start = 17, length = 21)

you could also expose a variable in the class that can be set to a lower value like 5 for the max items per call. so that you dont have to create these many items

Reply:
or, mock chunkArray function & return a stub object when called

---
In reply to: [309385066](https://github.com/microsoft/accessibility-insights-service/pull/222#discussion_r309385066) [](ancestors = 309385066)

Reply:
do not want to create var to expose, can test against 103 items, this is should not be the issue